enhancements/installer/openstack-baremetal-compute.md

```diff
@@ -0,0 +1,434 @@
+* Document how to use:
+  * specify two different worker sets in `install-config.yaml`: one for the VM
+    MachineSet, and one with Ironic-backed Nova flavors for the BM MachineSet
+
+#### Case 4. Install on VMs then add BM workers (BMs on a preexisting network)
+
+In this case, the cluster is installed on the installer-provisioned network.
+The additional MachineSet bears the `subnetName` of the preexisting network
+onto which the BMs will boot.
+
+Prerequisites for the user:
+
+* Connectivity must be ensured between the preexisting subnet and the cluster's
+  subnet (eg through a router).
+
+Expected work:
+
+* Allow configuring a Neutron subnet for a given worker node pool.
+* Adapt the Certificate-Signing-Request timeouts to the BM booting times.
+* Document how to use:
+  * make sure the `install-config.yaml` `MachineCIDR` contains the BMs network
+  * add a MachineSet with an Ironic-backed Nova flavor and the correct
+    `subnetName`.
+
+#### Adapt the wait-for install-complete timeout
+
+The last step of the installer is to check for the availability of the cluster
+and report a failed installation if a timeout is exceeded. However, BM nodes
+take longer than VMs to boot and may cause a timeout even in the case of a
+perfectly healthy cluster. With the proposed implementation of baremetal
+workers, the Installer is not aware whether the worker nodes are VMs or BMs; a
+conditional timeout extention is therefore not possible.
+
+The most straightforward solution would be to let the user set a custom timeout
+when installing, or disable the timeout altogether.
+
+#### Adapt the Certificate-Signing-Request timeouts to the BM booting times.
+
+In order to be authorised to join the cluster, new workers are expected to
+issue a CSR within a predefined timeout. This timeout currently might not
+account for BM boot times.
+
+#### Allow configuring a Neutron subnet for a given worker node pool
+
+We expect Machine API and [cluster-api-provider-openstack (CAPO)][capo] to be
+able to successfully create machines in a given subnet. Connectivity between
+the subnets is a responsibility we delegate to the user; however, CAPO must
+adapt the security groups to the new network topology.
+
+#### Handle VM to BM networking differences
+
+For every OpenStack worker VM, [CAPO][capo] always creates a Neutron port with
+the following configuration:
+
+* Security Groups
+* Allowed Address Pairs for the [Ingress VRRP IP
+  address][openstack-provider-vip]
+
+The worker node (Nova server) is requested to use this port rather than
+creating a new one.
+
+Depending on the ML2 driver these options may not be be available. We expect
+that in such cases the network traffic will not be blocked and the deployment
+will function as expected.
+
+We may however, need to add a more complex error recovery and fallback whereby
+we try to recreate the port without these features or possibly even create a
+server without adding an explicit port.
```

Comment:
It is probable, that we can add all of the requirements (security groups, address pairs, etc..) the port after the server is created.  in other words, we can create the server, then ask for what port it was attached to, then add the security groups and address pairs.